### PR TITLE
Various slight tweaks to README.md

### PR DESCRIPTION
**Pull request by LewsTherinTelescope**
Project looks awesome! I can't contribute much in terms of code (and the README says that that is discouraged for now anyway), but figured the least I could do was try to make some of the sentences in the README flow a bit better, since in my experience that's always a pain to refine when you want to be doing something more productive, hope that's alright! I've tried to add/remove as few words as possible, mostly reorganizing words and tweaking punctuation, but in some cases I've done so as long as it doesn't change the meaning or structure of the sentence at all.

### CODE REVIEW

**Comment by wilkomanger** [comment]
Thank you, this is really appreciated! It's an improvement! I'll add some comments for some minor thingies :+1:

File: README.md [hunk 1]
```diff
@@ -133,18 +133,16 @@ If you want to explicitly invoke a `const` constructor, you can use the followin
 Note the `@` before `const`. This is because `@const` is an annotation, not a keyword.
 The Kotlin compiler does not support keywords in front of expressions at the parser level.
 
-The difference is easy to remember, with any _declaration_ you must use `const`, and with any
+The difference is easy to remember: with any _declaration_ you must use `const`, and with any
 _invocation_ you must use `@const`.
 
-Note that `@const` is not necessary when it's implied by e.g. assigning to a `const val`,
-similar to Dart.
+Note that similar to in Dart, `@const` is not necessary when it's implied, e.g. by assigning to a `const val`.
```
Comment (wilkomanger) [inline README.md:139]:
```suggestion
Note that similar as in Dart, `@const` is not necessary when it's implied, e.g. by assigning to a `const val`.
```
I think this is more correct?

File: README.md [hunk 2]
```diff
@@ -235,10 +233,10 @@ Would throw the following error, because of the lambda literal argument:
 
 > ⚠️ Arguments of a constant creation must be constant expressions.
 
-Even though, if you'd pass a reference of a _named_ top-level/static function
+Even though if you'd passed a reference of a _named_ top-level/static function
```
Comment (wilkomanger) [inline README.md:236]:
```suggestion
Even though if you'd pass a reference to a _named_ top-level/static function
```
or
```suggestion
Even though if you would've passed a reference to a _named_ top-level/static function
```
Is correct I think :sweat_smile:

**Review by wilkomanger — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LewsTherinTelescope) [reply README.md:236]: Hmm, looking it up, I _think_ the way I currently have it ("if you had") is more "correct"? I think I've heard both ways before, though, so not sure.

Comment (wilkomanger) [reply README.md:236]: Ahh yeah "if you had" is correct indeed, currently it says "you'd" with the "'d" :smile:

Then I guess the third option is the following, which is what I think you were going for?
```suggestion
Even though if you passed a reference of a _named_ top-level/static function
```

File: README.md (hunk 1 above)
Comment (LewsTherinTelescope) [reply README.md]: Changed `similar to in` to just `as in`, which I think flows a bit better anyway.

**Review by wilkomanger — APPROVED** [review]

**Pull request merged**